cmd/notation/internal/display/metadata/json/inspect.go

```diff
@@ -0,0 +1,204 @@
+	}
+	return output.PrintPrettyJSON(h.printer, h.output)
+}
+
+// newSignature parses the signature blob and returns a Signature object.
+func newSignature(digest string, envelopeMediaType string, sigEnvelope signature.Envelope) (*Signature, error) {
+	envelopeContent, err := sigEnvelope.Content()
+	if err != nil {
+		return nil, err
+	}
+
+	signedArtifactDesc, err := envelope.DescriptorFromSignaturePayload(&envelopeContent.Payload)
+	if err != nil {
+		return nil, err
+	}
+
+	signatureAlgorithm, err := proto.EncodeSigningAlgorithm(envelopeContent.SignerInfo.SignatureAlgorithm)
+	if err != nil {
+		return nil, err
+	}
+	sig := &Signature{
+		MediaType:             envelopeMediaType,
+		Digest:                digest,
+		SignatureAlgorithm:    string(signatureAlgorithm),
+		SignedAttributes:      getSignedAttributes(envelopeContent),
+		UserDefinedAttributes: signedArtifactDesc.Annotations,
+		UnsignedAttributes:    getUnsignedAttributes(envelopeContent),
+		Certificates:          getCertificates(envelopeContent.SignerInfo.CertificateChain),
+		SignedArtifact:        *signedArtifactDesc,
+	}
+
+	// clearing annotations from the SignedArtifact field since they're already
+	// displayed as UserDefinedAttributes
+	sig.SignedArtifact.Annotations = nil
+
+	return sig, nil
+}
+
+func getSignedAttributes(envelopeContent *signature.EnvelopeContent) map[string]any {
+	signedAttributes := map[string]any{
+		"signingScheme": string(envelopeContent.SignerInfo.SignedAttributes.SigningScheme),
+		"signingTime":   envelopeContent.SignerInfo.SignedAttributes.SigningTime,
+	}
+	if expiry := envelopeContent.SignerInfo.SignedAttributes.Expiry; !expiry.IsZero() {
+		signedAttributes["expiry"] = expiry
+	}
+	for _, attribute := range envelopeContent.SignerInfo.SignedAttributes.ExtendedAttributes {
+		signedAttributes[fmt.Sprint(attribute.Key)] = fmt.Sprint(attribute.Value)
+	}
+	return signedAttributes
+}
+
+func getUnsignedAttributes(envelopeContent *signature.EnvelopeContent) map[string]any {
+	unsignedAttributes := make(map[string]any)
+	if envelopeContent.SignerInfo.UnsignedAttributes.SigningAgent != "" {
+		unsignedAttributes["signingAgent"] = envelopeContent.SignerInfo.UnsignedAttributes.SigningAgent
+	}
+	if envelopeContent.SignerInfo.UnsignedAttributes.TimestampSignature != nil {
+		unsignedAttributes["timestampSignature"] = parseTimestamp(envelopeContent.SignerInfo)
+	}
+	return unsignedAttributes
+}
+
+func getCertificates(certChain []*x509.Certificate) []Certificate {
+	var certificates []Certificate
+	for _, cert := range certChain {
+		hash := sha256.Sum256(cert.Raw)
+		certificates = append(certificates, Certificate{
+			SHA256Fingerprint: strings.ToLower(hex.EncodeToString(hash[:])),
+			IssuedTo:          cert.Subject.String(),
+			IssuedBy:          cert.Issuer.String(),
+			Expiry:            cert.NotAfter,
+		})
+	}
+	return certificates
+}
+
+func parseTimestamp(signerInfo signature.SignerInfo) Timestamp {
```

Comment:
In general, pointer should be returned.
```suggestion
func parseTimestamp(signerInfo signature.SignerInfo) *Timestamp {
```

Reply:
Updated. But why we need to return a pointer here but return struct in `envelope.DescriptorFromSignaturePayload`?